test/modules/http/compliance/http-res-raw-headers.test.ts

```diff
@@ -0,0 +1,32 @@
+import { it, expect, beforeAll, afterAll } from 'vitest'
+import http from 'http'
+import { ClientRequestInterceptor } from '../../../../src/interceptors/ClientRequest'
+import { waitForClientRequest } from '../../../helpers'
+
+const interceptor = new ClientRequestInterceptor()
+
+beforeAll(() => {
+  interceptor.apply()
+})
+
+afterAll(() => {
+  interceptor.dispose()
+})
+
+it('preserves the original mocked response headers casing in "rawHeaders"', async () => {
```

Comment:
I've also moved the raw headers test from `NodeClientRequest.test.ts` to a new compliance test. I think it makes sense to be here.